pkg/operator/certrotationcontroller/satokensigner_controller.go

```diff
@@ -274,6 +285,30 @@ func (c *SATokenSignerController) syncWorker() error {
 	return nil
 }
 
+// isValidKeyPair checks if a public/private keys match i.e if the tokens signed
+// with private key can be validated using public key. if "pubKeyData" or "privKeyData"
+// contains multiple keys, it tries to validate the first available keys in byte slices
+func isValidKeyPair(pubKeyData, privKeyData []byte) (bool, error) {
+	privKey, err := keyutil.ParsePrivateKeyPEM(privKeyData)
+	if err != nil {
+		return false, err
+	}
+	rsaPrivateKey, ok := privKey.(*rsa.PrivateKey)
+	if !ok {
+		return false, fmt.Errorf("private key is not of rsa type")
+	}
+	pubKeys, err := keyutil.ParsePublicKeysPEM(pubKeyData)
+	if err != nil {
+		return false, err
+	}
+	wantRSAPublicKey, ok := pubKeys[0].(*rsa.PublicKey)
```

Comment:
shouldn't it check any of them? can this include CAs?

Reply:
This is a public and private keypair. CAs are a tls concept, and are not involved here.

Reply:
And with only one public key and one private key